server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -387,8 +387,12 @@ private static Query tryRewriteLongSort(SearchContext searchContext, IndexReader
         ((sortField.getReverse() == false) && (missingValue == Long.MAX_VALUE));
         if (missingValuesAccordingToSort == false) return null;
 
+        int docCount = PointValues.getDocCount(reader, fieldName);
+         // is not worth to run optimization on small index, also estimation of duplicate data doesn't work well on small index
+        if (docCount <= 512) return null;
```

Comment:
I understand that we need a threshold but I don't understand this comment "estimation of duplicate data doesn't work well on small index", can you explain ?

Reply:
[BKDReader:estimatePointCount](https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/util/bkd/BKDReader.java#L633) returns `512` for small segments even if the whole segment only contains 100 docs.

Reply:
I am ok to rephrase this comment and also ok with another threshold

Reply:
Ok I got it now, thanks for explaining.